src/Converter_TEST.cc

```diff
@@ -618,6 +619,118 @@ TEST(Converter, RemoveDescendantElement)
   ASSERT_TRUE(convertedElem == nullptr);
 }
 
+////////////////////////////////////////////////////
+/// Ensure that Converter::Remove function is working
+/// Test removing empty elements only
+TEST(Converter, RemoveEmptyElement)
+{
+  // Set up an xml string for testing
+  std::string xmlString = getRepeatedXmlString();
+
+  // Verify the xml
+  tinyxml2::XMLDocument xmlDoc;
+  xmlDoc.Parse(xmlString.c_str());
+  tinyxml2::XMLElement *childElem =  xmlDoc.FirstChildElement();
+  ASSERT_NE(nullptr, childElem);
+  EXPECT_STREQ(childElem->Name(), "elemA");
+  childElem = childElem->FirstChildElement();
+  ASSERT_NE(nullptr, childElem);
+  EXPECT_STREQ(childElem->Name(), "elemB");
+  childElem = childElem->FirstChildElement();
+  ASSERT_NE(nullptr, childElem);
+  EXPECT_STREQ(childElem->Name(), "elemC");
+  childElem = childElem->FirstChildElement();
+  ASSERT_NE(nullptr, childElem);
+  EXPECT_STREQ(childElem->Name(), "elemD");
+
+  // Test removing empty elements
+  // Set up a convert file
+  std::stringstream convertStream;
+  convertStream << "<convert name='elemA'>"
+                << "  <convert name='elemB'>"
+                << "    <convert name='elemC'>"
+                << "      <remove_empty element='elemD'/>"
+                << "    </convert>"
+                << "  </convert>"
+                << "</convert>";
+  tinyxml2::XMLDocument convertXmlDoc;
+  convertXmlDoc.Parse(convertStream.str().c_str());
+  sdf::Converter::Convert(&xmlDoc, &convertXmlDoc);
+
+  tinyxml2::XMLElement *convertedElem =  xmlDoc.FirstChildElement();
```

Comment:
Can you add an assertion on `convertElem` here?

Reply:
https://github.com/ignitionrobotics/sdformat/pull/720/commits/4df59689dd05d4e8ff20029879777cfc143efacf